src/components/OptionsSelector.js

```diff
@@ -213,47 +225,101 @@ class OptionsSelector extends Component {
+            ? this.props.translate('common.confirm')
+            : this.props.confirmButtonText;
+        const defaultMaxParticipantsReachedMessage = _.isUndefined(this.props.maxParticipantsReachedMessage)
+            ? this.props.translate('common.maxParticipantsReached', {count: this.props.selectedOptions.length})
+            : this.props.maxParticipantsReachedMessage;
         return (
-            <View style={[styles.flex1]}>
-                <View style={[styles.ph5, styles.pv3]}>
-                    <TextInput
-                        ref={el => this.textInput = el}
-                        value={this.props.value}
-                        onChangeText={(text) => {
-                            if (this.props.shouldFocusOnSelectRow) {
-                                this.textInput.setNativeProps({selection: null});
-                            }
-                            this.props.onChangeText(text);
-                        }}
-                        onKeyPress={this.handleKeyPress}
-                        placeholder={this.props.placeholderText
-                            || this.props.translate('optionsSelector.nameEmailOrPhoneNumber')}
-                        onBlur={(e) => {
-                            if (!this.props.shouldFocusOnSelectRow) {
-                                return;
-                            }
-                            this.relatedTarget = e.relatedTarget;
-                        }}
-                        selectTextOnFocus
-                    />
-                </View>
-                <OptionsList
-                    ref={el => this.list = el}
-                    optionHoveredStyle={styles.hoveredComponentBG}
-                    onSelectRow={this.selectRow}
-                    sections={this.props.sections}
-                    focusedIndex={this.state.focusedIndex}
-                    selectedOptions={this.props.selectedOptions}
-                    canSelectMultipleOptions={this.props.canSelectMultipleOptions}
-                    hideSectionHeaders={this.props.hideSectionHeaders}
-                    headerMessage={this.props.headerMessage}
-                    disableFocusOptions={this.props.disableArrowKeysActions}
-                    hideAdditionalOptionStates={this.props.hideAdditionalOptionStates}
-                    forceTextUnreadStyle={this.props.forceTextUnreadStyle}
-                    showTitleTooltip={this.props.showTitleTooltip}
-                />
-            </View>
+            <ArrowKeyFocusManager
+                ref={el => this.focusManager = el}
+                listLength={this.props.canSelectMultipleOptions ? this.state.allOptions.length + 1 : this.state.allOptions.length}
+                onFocusedIndexChanged={this.scrollToIndex}
+                onEnterKeyPressed={this.selectFocusedIndex}
+                shouldEnterKeyEventBubble={focusedIndex => !this.state.allOptions[focusedIndex]}
```

Comment:
**question:** Why should the enter key event bubble if the option is missing?

Reply:
**thought:** If the `ArrowKeyFocusManager` had access to `allOptions` it would be able to deduce this?

Reply:
This prop is no longer here.